### PR TITLE
Stdevs

### PR DESCRIPTION
**Pull request by nick-Ag**
image

Finishes standard deviation feature and adds user defined periods for both standard deviation and running average. 
A new init-dialog component is added (image above) that is called by dataset.component and handles user input for the stats periods.

### CODE REVIEW

File: backend/server.py [hunk 1]
```diff
@@ -113,15 +115,26 @@ def compute_running_avg(trace, k=100):
     return avgs
     
 
-def compute_stdev(trace):
+def compute_stdev(trace, k=100):
```
Comment (rglasser) [inline backend/server.py:118]: Document k. Is there a better name that could be used? i/k/n starts to get confusing.

File: frontend/src/app/dataset/dataset.component.ts [hunk 2]
```diff
@@ -129,31 +178,74 @@ export class DatasetComponent {
...
+   * Handles all needed operations for requesting stats from the backend
+   * including sending a POST, and receiving and plotting the data.
+   * @param data The traces to send to the backend for which
+   *  stats will be computed.
+   * @param channel The data channel for which the stats were requested.
+   */
+  requestStats(data, channel) {
+    console.log('sending to server....', data);
+    this.sharedService.sendFormData(data, 'stats').subscribe((event: any) => {
+      if (
```
Comment (rglasser) [inline frontend/src/app/dataset/dataset.component.ts:206]: The spacing of this if block is very odd.

Comment (nick-Ag) [reply frontend/src/app/dataset/dataset.component.ts:206]: The inside of the block? or is the 'if' itself not tabbed over enough? The linter wants me to do it like that

Comment (rglasser) [reply frontend/src/app/dataset/dataset.component.ts:206]: The condition part is what's throwing me off. I feel like that should be indented more.

Comment (nick-Ag) [reply frontend/src/app/dataset/dataset.component.ts]: I tabbed them over more and added a /*eslint-disable*/

Comment (rglasser) [reply frontend/src/app/dataset/dataset.component.ts]: That's much more readable.

File: backend/server.py [hunk 3]
```diff
@@ -80,48 +80,62 @@ def compute_stats():
...
+    n = len(trace)
+    if period > n:
+        period = n
+
+    stdevs = [0 for i in range(n)]
+    for i in range(n):
+        if i < period:
+            stdevs[i] = np.std(trace[: i + 1])
+        else:
+            stdevs[i] = np.std(trace[(i + 1) - period : i + 1])
```
Comment (agarwal-v) [inline backend/server.py:136]: This is correct, but might be too slow for large data sets.

File: backend/server.py [hunk 4]
```diff
@@ -13,6 +13,7 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
+import time
```
Comment (agarwal-v) [inline backend/server.py:16]: sort imports...

File: frontend/src/app/plot/plot.component.ts [hunk 5]
```diff
@@ -149,6 +158,7 @@ export class PlotComponent {
    * @param id The id of the trace to toggle on/off.
    */
   toggleTrace(id: number) {
+    console.log('plot ids', this.idMap, ' id ', id);
```
Comment (agarwal-v) [inline frontend/src/app/plot/plot.component.ts:161]: If you are logging it, log its visibility as well

**Review by agarwal-v — COMMENTED** [review]
Minor nits

**Review by agarwal-v — APPROVED** [review]

**Pull request merged**